webapp/src/components/checklist_item/task_actions_modal.tsx

```diff
@@ -0,0 +1,234 @@
+    onTaskActionsChange: (newTaskActions: TaskActionType[]) => void,
+    taskActions?: TaskActionType[] | null,
+    playbookRunId?: string,
+) => ({
+    modalId: ID,
+    dialogType: TaskActionsModal,
+    dialogProps: {taskActions, onTaskActionsChange, playbookRunId},
+});
+
+export const keywordsTriggerEmptyPayload = {keywords: [] as string[], user_ids: [] as string[]};
+export const markAsDoneEmptyPayload = {enabled: false};
+
+type Props = {
+    onTaskActionsChange: (newTaskActions: TaskActionType[]) => void,
+    taskActions?: TaskActionType[] | null,
+    playbookRunId?: string,
+} & Partial<ComponentProps<typeof GenericModal>>;
+
+const TaskActionsModal = ({onTaskActionsChange, taskActions, playbookRunId, ...modalProps}: Props) => {
+    const {formatMessage} = useIntl();
+    const emptyTask = {} as TaskActionType;
+    const taskAction = (taskActions && (taskActions.length > 0)) ? taskActions[0] : emptyTask;
+
+    // we only have one kind of trigger
+    const triggerType = taskAction.trigger?.type ? taskAction.trigger.type : KeywordsByUsersTriggerType;
+    const triggerPayload = taskAction.trigger?.payload ? JSON.parse(taskAction.trigger.payload) : keywordsTriggerEmptyPayload;
+
+    // we currently only support one action per trigger
+    const actionType = (taskAction.actions?.length > 0 && taskAction.actions[0]?.type) ? taskAction.actions[0].type : MarkItemAsDoneActionType;
+    const actionPayload = (taskAction.actions?.length > 0 && taskAction.actions[0]?.payload) ? JSON.parse(taskAction.actions[0].payload) : markAsDoneEmptyPayload;
+
+    const [show, setShow] = useState(true);
+    const users = useSelector(getUsers);
+    const defaultUsers = triggerPayload.user_ids.map((user_id: string) => users[user_id]);
+
+    const [newKeywords, setNewKeywords] = useState(triggerPayload.keywords);
+    const [newUserIDs, setNewUserIDs] = useState(triggerPayload.user_ids);
+    const [newIsEnabled, setNewEnabled] = useState(actionPayload.enabled);
+
+    const runID = playbookRunId || '';
+    const run = useSelector(getRun(runID));
+    const teamID = useSelector(getCurrentTeamId);
+    let searchOpts = {} as any;
+    if (playbookRunId) {
+        searchOpts = {in_channel_id: run?.channel_id};
+    } else {
+        searchOpts = {team_id: teamID};
+    }
+    const dispatch = useDispatch();
+    const searchUsers = (term: string) => {
+        return dispatch(searchProfiles(term, searchOpts));
+    };
+
+    const onSave = () => {
+        const newTaskAction = {} as TaskActionType;
+        const newTriggerPayload = {keywords: newKeywords, user_ids: newUserIDs};
+        newTaskAction.trigger = {
+            type: KeywordsByUsersTriggerType,
+            payload: JSON.stringify(newTriggerPayload),
+        };
+
+        const newActionPayload = {enabled: newIsEnabled};
+        newTaskAction.actions = [{
+            type: MarkItemAsDoneActionType,
+            payload: JSON.stringify(newActionPayload),
+        }];
```

Comment:
To reduce type casting, improve readability IMHO (2/5)

```suggestion
        const newTaskAction: TaskAction = {
            trigger: {
                type: KeywordsByUsersTriggerType,
                payload: JSON.stringify({keywords: newKeywords, user_ids: newUserIDs}),
            },
            actions: [
                {
                    type: MarkItemAsDoneActionType,
                    payload: JSON.stringify({enabled: newIsEnabled}),
                },
            ],
        };
```